File: pset6/cash.py
from cs50 import get_float

quarter = 25
dime = 10
nickel = 5
penny = 1
coins = 0

while True:
    f = get_float("Changed owed: ")
    if f > 0:
        break
    
# Converts Dollars into Cents
cents = round(f * 100)

# Counts the cents via division + works out any remainder to pass on
coins += cents // quarter
cents = (cents % quarter)
    
# Counts the dimes via division + works out any remainder to pass on
coins += cents // dime
cents = (cents % dime)

# Counts the nickels via division + works out any remainder to pass on
coins += cents // nickel
cents = (cents % nickel)

# Counts the pennys via division + works out any remainder to pass on
coins += cents // penny
cents = (cents % penny)
    
print(coins)

File: pset1/cash.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int main(void)
{
    int quarter = 25;
    int dime = 10;
    int nickel = 5;
    int penny = 1;
    float f;
    int coins = 0;

    //Sets a loop to continuiously check for a valid float
    do
    {
        f = get_float("Changed owed: ");
    }
    while (f <= 0);

    //Rounds up and converts "dollars" into "cents"
    int cents = round(f * 100);
    while (cents >= quarter)
    {
        coins += cents / quarter;
        cents = (cents % quarter);
    }
    while (cents >= dime)
    {
        coins += cents / dime;
        cents = (cents % dime);
    }

    while (cents >= nickel)
    {
        coins += cents / nickel;
        cents = (cents % nickel);
    }

    while (cents >= penny)
    {
        coins += cents / penny;
        cents = (cents % penny);
    }
    printf("%i\n", coins);

}